scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,118 @@
+"""Module to generate a column map to their corresponding statistical variable defintions"""
+
+import os
+import json
+import pandas as pd
+
+#TODO:Remove this import and make it from ~/data/utils/ directory when it is available
+os.system('curl https://raw.githubusercontent.com/Abilityguy/data/dcid_gen/util/statvar_dcid_generator.py -LO statvar_dcid_generator.py')
+import statvar_dcid_generator
+
+dcid_gen = statvar_dcid_generator.StatVarDcidGenerator()
+
+
+def generate(specDict, columnList):
+  """Wrapper function for generateColMapBase class to generate column map"""
+  col_map_obj = generateColMapBase(specDict=specDict, columnList=columnList)
+  return col_map_obj._generate_stat_vars_from_spec()
+
+class generateColMapBase:
+  """module to generate a column map given a list of columns of the dataset and a JSON Spec"""
+  def __init__(self, specDict=None, columnList=None):
+    """module init """
+    self.features = specDict
+    self.column_list = columnList
+    self.column_map = {}
+
+  def _generate_stat_vars_from_spec(self):
+    """generates stat_var nodes for each column in column list and returns a dictionary of columns with their respective stat_var nodes"""
+    # for each column generate the definition of their respective statistical variable node
+    for col in self.column_list:
+      if 'ignoreColumns' in self.features and col not in self.features['ignoreColumns']:
+          self.column_map[col] = self._column_to_statvar(col)
+    #TODO: Enable validating the generate columnMap: _valid_col_map():
+    return self.column_map
+
+  def _column_to_statvar(self, column):
+    """generates a dictionary statistical variable with all properties specified in the JSON spec for a single column"""
+    measurement_assigned = False
+    stat_var = {}
+    # part_list contains a list of tokens of the column name after splitting the
+    # column name based on the '!!' delimiter
+    part_list = column.split('!!')
+    # part_list[-1], the last element of part_list describes the property value.
+    # This element points to the 'v' in 'pv' -> and we map it to the
+    # corresponding property 'p'
+    val_prop_col = part_list[-1]
```

Comment:
Is this always true (aka some census documentation or something that confirms this)?

Rather I wonder what is wrong if we exact match every column part against every C value in pvs section?

Reply:
I came to this assumption based on two factors: observations on the column name string and the [ACS Subject Table shells](https://www2.census.gov/programs-surveys/acs/tech_docs/table_shells/2019/). Will this suffice? 
For this, would a few examples from the dataset be helpful?

Reply:
For example: https://screenshot.googleplex.com/7tgoYGjhyoRrgba ... that's gender, age and one more PV.

Granted, this is a B Table, but it might be good to not make constraining assumptions, especially when we don't really need to.

Reply:
Based on the earlier feedback, I had removed the use of this assumption and this line was something that should have been removed but was not. It is not used anywhere in the code.